integration/benchmark/cmd/ci/main.go

```diff
@@ -163,57 +170,56 @@ func main() {
 	if err != nil {
 		log.Fatal().Err(err).Msgf("unable to init loader")
 	}
+
+	// run load
 	lg.Start()
 
-	testStartTime := time.Now()
+	// prepare data slices
+	dataSlices := make([]dataSlice, 0)
+	go func() {
```

Comment:
you may want to factor it out into a function (it'll be easier to both read and unnittest it).